server/api/roster/roster-entity.ts

```diff
@@ -79,3 +216,14 @@ export const baseRosterColumns: RosterColumnInfo[] = [
     updatable: true,
   },
 ];
+
+export type RosterEntryData = {
+  edipi: RosterEntity['edipi'],
+  unit: number,
+  firstName?: RosterEntity['firstName'],
+  lastName?: RosterEntity['lastName'],
+} & CustomColumns;
+
+export type RosterFileRow = {
```

Comment:
I almost always prefer the non-Record version, just because it lets you give context to the key (not that the current chunk is a good example of that). My preference would be for it to be like this...

```typescript
export type RosterFileRow = {
  [columnName: string]: string
};
```